src/riak_kv_eleveldb_backend.erl

```diff
@@ -478,17 +493,19 @@ range_scan(FoldIndexFun, Buffer, Opts, #state{fold_opts=_FoldOpts,
             [] -> AdditionalOptions;
             _  -> [{range_filter, Filter} | AdditionalOptions]
         end,
-    StartK2 = [{Field, Val} || {Field, _Type, Val} <- StartK],
+    %% Pad with missing local keys after the timestamp - use minimum erlang term value of '0' to make
+    %% sure we start the search at the beginning.
+    StartK2 = [{Field, Val} || {Field, _Type, Val} <- StartK] ++ [{N, 0} || #param_v1{name = [N]} <- ExtraLK],
     StartK3 = riak_ql_ddl:make_key(Mod, LK, StartK2),
     StartK4 = riak_kv_ts_util:encode_typeval_key(StartK3), %% TODO: Avoid adding/removing type info
     StartKey = to_object_key(Bucket, StartK4),
-    EndK2 = [{Field, Val} || {Field, _Type, Val} <- EndK],
+    %% Pad the missing end - not perfect because there could be longer bitstrings, but will have to do for prototype
+    %% TODO: FIX END KEY
+    EndK2 = [{Field, Val} || {Field, _Type, Val} <- EndK] ++ [{N, <<16#ffffffff:64>>} || #param_v1{name = [N]} <- ExtraLK],
```

Comment:
There's still the issue of the end key to be addressed - if the end key had 65 set bits it would be missed.

The problem is Erlang Term ordering goes like this
number < atom < reference < fun < port < pid < tuple < map < nil < list < bit string

So if you're trying to find a term that will be after all of the other terms, doing it this naive way you would need an infinitely long stream of bits.  I think you could do it by converting the binary back to an integer, incrementing by 1 and converting back to a binary, but you would have to handle overflow cases.